provisioning/provisioning_info.go

```diff
@@ -18,7 +19,8 @@ const (
 )
 
 type ProvisioningInfo struct {
-	Client             kubernetes.Interface
+	KubeClient         kubernetes.Interface
```

Comment:
AFAICS, KubeClient is used 4 times and Client is used frequently. Isn't it possible to use one of them(Possibly `client.Client`) to uniform client usage?. Otherwise, I saw a couple of upstream repositories that there are lots of different client objects and contributors seem like using them randomly(or I don't understand the procedure which one is used in which cases :) ).

Reply:
i could try and separate them for clarity. but the reason I am keeping them separate is the r.Client is a cached client for the namespace we watching. but if we access things explicitly in other namespaces i believe we still need to access it using the kubeclient. but i'll test to make sure (i would have thought it would work fine, it would just have read from the API).
For the webhook i have to use kubeclient as we need to use it before the cached client is initialised.